ansible/deploy-openstack-config.yml

```diff
@@ -0,0 +1,184 @@
+        - { src: configure-vxlan.yml, dest: seed-host-configure/pre.d/20-configure-vxlan.yml }
+
+
+    - name: Ensure Admin Overcloud Network file is present
+      ansible.builtin.copy:
+        src: "files/admin-oc-networks.yml"
+        dest: "{{ src_directory }}/{{ kayobe_config_name }}/etc/kayobe/environments/{{ kayobe_config_environment }}/admin-oc-networks.yml"
+        mode: "0644"
+
+    - name: Ensure Hosts file is present
+      ansible.builtin.copy:
+        src: "files/hosts"
+        dest: "{{ src_directory }}/{{ kayobe_config_name }}/etc/kayobe/environments/{{ kayobe_config_environment }}/inventory/hosts"
+        mode: "0644"
+
+    - name: Ensure Kayobe repository is present
+      ansible.builtin.git:
+        repo: "{{ kayobe_repo }}"
+        version: "{{ kayobe_version }}"
+        dest: "{{ src_directory }}/{{ kayobe_name }}"
+        update: false
+
+    - name: Ensure `venvs` directory exists
+      ansible.builtin.file:
+        path: "{{ ansible_env.HOME }}/venvs"
+        state: directory
+        mode: "0755"
+
+    - name: Ensure `kayobe` is present
+      ansible.builtin.pip:
+        name: "{{ src_directory }}/{{ kayobe_name }}/"
+        virtualenv: "{{ ansible_env.HOME }}/venvs/kayobe"
+        state: present
+
+    - name: Ensure vault password is present
+      ansible.builtin.copy:
+        src: "{{ vault_password_path }}"
+        dest: "{{ ansible_env.HOME }}/vault-pw"
+        mode: "0600"
+
+    - name: Ensure public key is present
+      ansible.builtin.copy:
+        src: "{{ public_key_path }}"
+        dest: "{{ ansible_env.HOME }}/.ssh/id_rsa.pub"
+        mode: "0644"
+
+    - name: Ensure private key is present
+      ansible.builtin.copy:
+        src: "{{ private_key_path }}"
+        dest: "{{ ansible_env.HOME }}/.ssh/id_rsa"
+        mode: "0600"
+
+    - name: Ensure OpenStack Config repository is present
+      ansible.builtin.git:
+        repo: "{{ openstack_config_repo }}"
+        dest: "{{ src_directory }}/{{ openstack_config_name }}"
+        version: "{{ openstack_config_version }}"
+        update: false
+
+    - name: Ensure `openstack` virtualenv is present
+      ansible.builtin.pip:
+        requirements: "{{ src_directory }}/{{ openstack_config_name }}/requirements.txt"
+        virtualenv: "{{ src_directory }}/{{ openstack_config_name }}/ansible/openstack-config-venv"
+        state: present
+
+    - name: Ensure `openstack` ansible role requirements are present
+      community.general.ansible_galaxy_install:
+        dest: "{{ src_directory }}/{{ openstack_config_name }}/ansible/roles"
+        requirements_file: "{{ src_directory }}/{{ openstack_config_name }}/requirements.yml"
+        type: role
+      vars:
+        ansible_python_interpreter: "{{ src_directory }}/{{ openstack_config_name }}/ansible/openstack-config-venv/bin/python3"
+      environment:
+        PATH: "{{ ansible_env.HOME }}/venvs/openstack/bin/:{{ ansible_env.PATH }}"
```

Comment:
Where does this venv get created? Maybe this one would be better to create above

Reply:
Fixed. Was left over from an earlier commit that had openstack's virtualenv created alongside kayobe's